src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -594,12 +604,18 @@ DistributedInsertSelectSupported(Query *queryTree, RangeTblEntry *insertRte,
 		return error;
 	}
 
-	/*
-	 * If we're inserting into a reference table, all participating tables
-	 * should be reference tables as well.
-	 */
-	if (IsCitusTableType(targetRelationId, REFERENCE_TABLE))
+	if (IsCitusTableType(targetRelationId, CITUS_LOCAL_TABLE))
 	{
+		/*
+		 * If we're inserting into a citus local table, it is ok.
```

Comment:
it is ok because we've checked the non-existence  of distributed tables in the subquery